docs/metrics.md

```diff
@@ -0,0 +1,59 @@
+# Metrics
+
+Kiam exports both Prometheus and StatsD metrics to determine the health of the
+system, check the timing of each RPC call, and monitor the size of the
+credentials cache. By default, Prometheus metrics are exported on
+`localhost:9620` and StatsD metrics are sent to `127.0.0.1:8125`. StatsD
+metrics are not aggregated and flushed every 100ms.
+
+## Metrics configuration
+
+- The `statsd` flag controls the address to which to send StatsD metrics. This
+  is by default `127.0.0..1:8125`. If this is blank, StatsD metrics will be
+  silenced.
+- The `statsd-prefix` flag controls the initial prefix that will be appended to
+  Kiam's StatsD metrics. This is by default `kiam`.
+- The `statsd-interval` flag controls how frequently the in-memory metrics
+  buffer will be flushed to the specified StatsD endpoint. Metrics are
+  not aggregated in this buffer and the raw counts will be flushed to the
+  underlying StatsD sink. This is by default `100ms`.
+- The `prometheus-listen-addr` controls which address Kiam should create a
+  Prometheus endpoint on. This is by default `localhost:9620`. The metrics
+  themselves can be accessed at `<prometheus-listen-addr>/metrics`.
+- The `prometheus-sync-interval` flag controls how frequently Prometheus
+  metrics should be updated. This is by default `5s`.
+
+## Emitted Metrics
+
+### Prometheus
+#### Metadata Subsystem
+- `handler_latency_milliseconds` - Bucketed histogram of handler timings. Tagged by handler
+- `credential_fetch_error` - Number of errors fetching the credentials for a pod
```

Comment:
Should this be renamed to `credential_fetch_errors_total` to indicate its an accumulating counter?